src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -125,6 +125,14 @@ private BoundStatement RewriteEnumeratorForEachStatement(BoundForEachStatement n
             //     V v = (V)(T)e.Current;
             //     /* node.Body */
             // }
+
+            // The scope of the iteration variable is the foreach statement syntax.
```

Comment:
They absolutely help me to implement EnC. IsClosureScope is called from the IDE to find out what syntax corresponds to display classes (BoundBlocks). So the assertion is making sure that the IDE and the compiler are in sync. I agree we might want to consider improving the way how to make this information available to the IDE in future (perhaps exposing some public APIs, semantic trees, etc.). We can revisit this for C# 7. For the current version I'm not gonna change the fundamentals of the EnC infrastructure.